src/hmm/transition-model.h

```diff
@@ -141,19 +141,21 @@ class TransitionModel {
   /// \name Integer mapping functions
   /// @{
 
-  int32 TripleToTransitionState(int32 phone, int32 hmm_state, int32 pdf) const;
+  int32 TupleToTransitionState(int32 phone, int32 hmm_state, int32 pdf, int32 self_loop_pdf) const;
   int32 PairToTransitionId(int32 trans_state, int32 trans_index) const;
   int32 TransitionIdToTransitionState(int32 trans_id) const;
   int32 TransitionIdToTransitionIndex(int32 trans_id) const;
   int32 TransitionStateToPhone(int32 trans_state) const;
   int32 TransitionStateToHmmState(int32 trans_state) const;
   int32 TransitionStateToPdf(int32 trans_state) const;
+  int32 TransitionStateToSelfLoopPdf(int32 trans_state) const;
   int32 SelfLoopOf(int32 trans_state) const;  // returns the self-loop transition-id, or zero if
   // this state doesn't have a self-loop.
 
   inline int32 TransitionIdToPdf(int32 trans_id) const;
   int32 TransitionIdToPhone(int32 trans_id) const;
   int32 TransitionIdToPdfClass(int32 trans_id) const;
```

Comment:
Just noticed this (after debugging)... this function [TransitionIdToPdfClass] needs to check whether it's a self loop and return a value accordingly.  I.e. it should return the; and there should probably be no function TransitionIdToSelfLoopPdfClass [its existence makes things a bit unclear; and I doubt there would be a legitimate use for it.]